cypress-io/cypress
# chore: updated styling for configuration files

- Closes #21847 

### User facing changelog

- In order to improve styling consistency on the Cypress app, we need to ensure that panels on migration and configuration have consistent border outlining. (Check #21847 for styling differences between migration and configuration file panels)
- According to both [Figma](https://www.figma.com/file/1WJ3GVQyMV5e7xVxPg3yID/Design-System?node-id=2232%3A3729) and the [design system components](https://design.cypress.io/?path=/docs/react_accordion--accordion), only the header should have the border. This PR addresses this by ensuring that only the headers on "configuration files" accordions have that styling and not the entire panel. 

### Additional details

- Why was this change necessary? So panels on different screens e.g. migration and configuration have similar styling for consistency.
- FileRow.vue (Which renders the panels on the configuration) uses Collapsible.vue. Since Collapsible.vue is a reusable component that is leveraged by other components e.g. SettingCard, GeneratorSuccess e.t.c., adding any new styles to Collapsible for the purpose of re-designing FileRow would also adversely impact the other components that use Collapsible as unwanted styles would be applied to those components as well. In order to prevent this, I have added an optional prop in Collapsible, which when set to true would apply changes to the Collapsible component. The other components would not pass in this prop and therefore will not receive any unwanted styling changes.
- In order for the header section of the panel be highlighted we need to independently control styles for the header and the bottom section of the panel. This is only possible by updating styles on Collapsible.vue.

### Steps to test

- No new tests have been added because only the UI/ styling for the configuration page has changed. All the functionality of the component has stayed the same.
- However, in order to look at the updated UI, you can:
-
1. Run yarn dev from the root folder (cypress)
2. Add the launchpad as a project on the Cypress UI
3. Go to e2e testing and run the project-setup.cy.ts spec file. Check the 'opens correctly in unconfigured project with --e2e' test to see the updated look of the config files.
4. Change to Component Testing and run the FileRow.cy.ts spec file. Check the 'renders each status, the expected files, and expected styles' to see the updated look on different types of config files.

### How has the user experience changed?

Before:
<img width="785" alt="Screen Shot 2022-09-12 at 11 16 25 AM" src="https://user-images.githubusercontent.com/65267668/189691609-2159b325-0cbb-44d5-8d94-9099dded68c7.png">
After:
<img width="785" alt="Screen Shot 2022-09-12 at 11 21 34 AM" src="https://user-images.githubusercontent.com/65267668/189692716-608c9bed-1a27-44c9-88dc-60c67b2264eb.png">


### PR Tasks


- [na] Have tests been added/updated?
- [na] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
